framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml

```diff
@@ -305,18 +305,6 @@ sWscAttrKeyWrapAuthenticator:
 
 #encrypted attributes
 
-sWscAttrSsid:
-  _type: struct
-  attribute_type:
-    _type: eWscAttributes
-    _value: ATTR_SSID
-  data_length:
-    _type: uint16_t
-    _value: WSC_MAX_SSID_LENGTH
-  data:
-    _type: char
-    _length: [ WSC_MAX_SSID_LENGTH ]
```

Comment:
Actually, I think it would be better to change cConfigData to make use of this attribute. The attribute definition is wrong, though, length should be either `[]` or `[ data_length ]`. Oh, and that means it would have to become a class instead of a struct...

Note also that a lot of other attributes suffer from the same shortcoming (they're declared as fixed length instead of dynamic length).

BTW the commit message is wrong, it talks about M1 and M2 but it's actually used in cConfigData.

Obviously, this requires the callers to change as well, and *that* must be done in the same commit as the yaml change.

Reply:
You are right but unfortunately, if my memory serves me right, we don't support variable length lists inside structs inside classes, I can't find an issue for some reason, we must have closed it.
This is also the reason that all the string attributes (such as manufacturer, device name, etc.) are put as is in the M2 and M1 TLVs and not using the attributes.
This should be fixed obviousely since it make the YAML much longer - but not a high priority.